src/os/windows/font_win32.cpp

```diff
@@ -233,54 +233,29 @@ static std::string GetEnglishFontName(const ENUMLOGFONTEX *logfont)
 }
 #endif /* WITH_FREETYPE */
 
-class FontList {
-protected:
-	wchar_t **fonts;
-	uint items;
-	uint capacity;
-
-public:
-	FontList() : fonts(nullptr), items(0), capacity(0) { };
-
-	~FontList() {
-		if (this->fonts == nullptr) return;
-
-		for (uint i = 0; i < this->items; i++) {
-			free(this->fonts[i]);
-		}
-
-		free(this->fonts);
-	}
+struct EFCParam {
+	FontCacheSettings *settings;
+	LOCALESIGNATURE  locale;
+	MissingGlyphSearcher *callback;
+	std::vector<std::unique_ptr<wchar_t>> fonts;
 
 	bool Add(const wchar_t *font) {
-		for (uint i = 0; i < this->items; i++) {
-			if (wcscmp(this->fonts[i], font) == 0) return false;
-		}
-
-		if (this->items == this->capacity) {
-			this->capacity += 10;
-			this->fonts = ReallocT(this->fonts, this->capacity);
+		for (const auto &entry : this->fonts) {
+			if (wcscmp(entry.get(), font) == 0) return false;
 		}
 
-		this->fonts[this->items++] = wcsdup(font);
+		this->fonts.emplace_back(wcsdup(font));
```

Comment:
Essentially you're mixing `malloc` with `delete` here due to `unique_ptr`. What about `std::vector<std::wstring>` and getting rid of `wcsdup`?
Maybe even `std::wstring_view &font` and the C++ compare between `entry.get()` and `font`?

Reply:
Oops, stupid. Yes, `wstring`/`wstring_view` work here.